extension-push/src/java/com/defold/push/LocalNotificationReceiver.java

```diff
@@ -25,13 +28,29 @@ public void onReceive(Context context, Intent intent) {
             // If activity is visible we can just send data to the listener without intent
             Push.getInstance().onLocalPush(context, extras.getString("payload"), id, false);
         } else {
-            Notification notification = intent.getParcelableExtra(context.getPackageName() + Push.DEFOLD_NOTIFICATION);
+            String packageName = context.getPackageName();
+            Notification notification = intent.getParcelableExtra(packageName + Push.DEFOLD_NOTIFICATION);
             if (notification != null) {
+                NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(context, notification);
+                int smallIconId = context.getResources().getIdentifier("push_icon_small", "drawable", packageName);
+                if (smallIconId == 0) {
+                    ApplicationInfo info = context.getApplicationInfo();
+                    smallIconId = info.icon;
+                    if (smallIconId == 0) {
+                        smallIconId = android.R.color.transparent;
+                    }
+                }
+                notificationBuilder.setSmallIcon(smallIconId);
+                if (Build.VERSION.SDK_INT >= 23) {
+                    notificationBuilder.setSmallIcon(IconCompat.createWithResource(context, smallIconId));
+                }
                 nm = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
-                nm.notify(id, notification);
+                Notification newNotification = notificationBuilder.build();
```

Comment:
I wonder if we shouldn't move some of this code to a shared place? We have the Notification created here and in two places in Push.java

Also getting and setting the icons above?

Reply:
there is no Draft PR in this repo.
Sorry, I forgot to mention this is work in progress PR to keep track this issue 

there are two ways to fix that:
1. re-create notification as in this PR
2. use special xml files where icon resource id specified and aapt tool uses it from there.

I will investigate both options

Reply:
hmmm, yes we do it 3 times, but I don't see any way to re-use it